vision/Unittest.py

```diff
@@ -0,0 +1,38 @@
+import unittest
+from ModuleKMeans import ModuleKMeans as mkm
+
+class TestKMeans(unittest.TestCase):
+
+    def test_fractal(self):
+        """
+        Testing Fractal KMeans Display
+
+        Test images must be put in a file called images
+        """
+
+        for pic in ["images/blocks1.jpg",
+                    "images/blocks2.jpg",
+                    "images/pylon.jpg"]:
+            img = mkm(pic)
+            img.applyKMeans(3)
+            channels = [0, 1]
+            img.displayFractal()
+
+    def test_binary(self):
+        """
+        Testing Binary KMeans Display
+
+        Test images must be put in a file called images
+        """
+        
+        for pic in ["images/blocks1.jpg",
+                    "images/blocks2.jpg",
+                    "images/pylon.jpg"]:
+            img = mkm(pic)
+            img.applyKMeans(3)
+            channels = [0, 1]
+            img.displayBinary(channels)
```

Comment:
Both of these tests require a human to interpret the results, which is okay, but should not be done here as a unit test. Id recommend putting both of these functions in the `if __name__ == '__main__':` of the  KMeans file. We do want to run a unit test on the clustering algorithm and make sure it responds to parameter changes as it should, but without a tool to generate a goal output for each image this is hard. For now you should find an image that your algorithm works well on and save the output, the unit test should ensure that future output resembles it within some error margin. This way whoever creates the tool for us will have something to work with.